janitor/functions.py

```diff
@@ -3150,7 +3150,19 @@ def update_where(
     :raises: IndexError if **conditions** does not have the same length as
         **df**.
     """
-    df.loc[conditions, target_column_name] = target_val
+    # use query style method
+    # to accommodate changes if the dataframe has been modified
+    # somewhere along the method chain block
+    # suggested by @zbarry
+
+    if not isinstance(conditions, str):
+        raise TypeError("conditions argument should be an expression")
```

Comment:
Will rejecting boolean statements introduce a breaking change into the `update_where` function? If so, the deprecation has to be managed carefully. Step 1 is enabling both to live side by side for a few versions, leaving a comment and raising an issue, and then Step 2, a few versions later, remove the deprecated version.